pkg/backup/compaction_processor.go

```diff
@@ -374,17 +385,15 @@ func openSSTs(
 	if err != nil {
 		return mergedSST{}, err
 	}
+	mergedSSTDirs := dirs
```

Comment:
I find this swapping of dirs quite confusing. What about the following instead? After refactoring the return args to be named?
```
cleanupDirs := func() {
		for _, dir := range dirs {
			if err := dir.Close(); err != nil {
				log.Warningf(ctx, "close export storage failed: %v", err)
			}
		}
	}

defer () {
  // err here is the named return err
  if err != nil {
    cleanupDirs()
  }
}

```

Reply:
Oh that's what I had originally, but you end up having to be pretty careful about the closures to make sure you set the right `err`. What are your thoughts on that tradeoff?